convert.py

```diff
@@ -73,21 +73,46 @@
         logging.error(f"Mapper failed with code {exit_code}. Exiting.")
         exit(exit_code)
 
-    ### Convert model to text format so we can read cameras
-    convert_cmd = (colmap_command + " model_converter \
-        --input_path " + args.source_path + "/distorted/sparse/0 \
-        --output_path "  + args.source_path + "/distorted/sparse/0 \
-        --output_type TXT")
-    exit_code = os.system(convert_cmd)
-    if exit_code != 0:
-        logging.error(f"Convert failed with code {exit_code}. Exiting.")
-        exit(exit_code)
+
+# select the largest submodel
+i = 0
+largest_size = 0
+index = 0
+
+while True:
+    path = args.source_path + "/distorted/sparse/" + str(i)
+    if not os.path.exists(path):
+        break
+
+    # check the file size of images.bin
+    images_bin = path + "/images.bin"
+    size = os.path.getsize(images_bin)
+    if size > largest_size:
+        largest_size = size
+        index = i
+
+    i += 1
+
+str_index = str(index)
+distorted_sparse_path = args.source_path + "/distorted/sparse/" + str_index
+
+
+### Convert model to text format so we can read cameras
+convert_cmd = (colmap_command + " model_converter \
```

Comment:
Currently gs-generation/app.js extracts cameras from `/distorted/sparse/0/images.txt`. With this change that won't work anymore. I'd suggest specifying output location of `/distorted/sparse/` here and load then load cameras from `/distorted/sparse/images.txt` instead.

Reply:
it's updated in monorepo